main.go

```diff
@@ -58,81 +60,135 @@ type config struct {
 	upstreamForceH2C      bool
 	upstreamCAFile        string
 	auth                  proxy.Config
-	tls                   tlsConfig
 	kubeconfigLocation    string
 	allowPaths            []string
 	ignorePaths           []string
+	configFileName        string
+	ServingOptions        *genericapiserveroptions.SecureServingOptionsWithLoopback
+	AuthenticationOptions *genericapiserveroptions.DelegatingAuthenticationOptions
 }
 
-type tlsConfig struct {
-	certFile       string
-	keyFile        string
-	minVersion     string
-	cipherSuites   []string
-	reloadInterval time.Duration
-}
-
-type configfile struct {
-	AuthorizationConfig *authz.Config `json:"authorization,omitempty"`
-}
-
-func main() {
-	cfg := config{
+func NewConfig() *config {
+	return &config{
 		auth: proxy.Config{
 			Authentication: &authn.AuthnConfig{
-				X509:   &authn.X509Config{},
 				Header: &authn.AuthnHeaderConfig{},
 				OIDC:   &authn.OIDCConfig{},
 				Token:  &authn.TokenConfig{},
 			},
 			Authorization: &authz.Config{},
 		},
+		ServingOptions:        genericapiserveroptions.NewSecureServingOptions().WithLoopback(),
+		AuthenticationOptions: genericapiserveroptions.NewDelegatingAuthenticationOptions(),
 	}
-	configFileName := ""
-
-	// Add klog flags
-	klogFlags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
-	klog.InitFlags(klogFlags)
+}
 
-	flagset := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
-	flagset.AddGoFlagSet(klogFlags)
+func (c *config) AddFlags(flags *pflag.FlagSet) {
```

Comment:
no vanity renames in this PR.  Don't use flags, use flagset.  Don't use c, use cfg

Reply:
agreed